lib/src/test/groovy/com/cloudbees/groovy/cps/impl/FunctionCallBlockTest.groovy

```diff
@@ -21,6 +22,28 @@ class FunctionCallBlockTest extends AbstractGroovyCpsTest {
             throw new NamingException();
     }
 
+    /** Need to run more steps so we can hit the recursion limit */
+    Object evalCPSonly(String script, int numSteps) {
+        return parseCps(script).invoke(null, null, Continuation.HALT).run(numSteps).replay()
+    }
+
+    @Test
+    void infiniteRecursion() {
+        try {
+            assert evalCPSonly("""
+                def thing = null;
+                def getThing() { 
+                    return thing == null;
+                }
+                def stuff = getThing();
```

Comment:
Cf. the associated issue. It seems this _is_ a buggy program (reproducible using command-line Groovy—`def thing` does not declare a field), but it might be clearer for test purposes to use a self-evidently endless recursion, such as:

```groovy
def buggy() {
    buggy()
}
buggy()
```

Reply:
i don’t think def thing is needed at all to show this anyway.